### PR TITLE
[Merged by Bors] - Add `get_directory_group` Operation to `DirectorySync` module

### PR DESCRIPTION
**Pull request by sheldonvaughn**

### CODE REVIEW

File: src/directory_sync/operations/get_directory_group.rs [hunk 1]
```diff
@@ -0,0 +1,142 @@
+use async_trait::async_trait;
+use reqwest::StatusCode;
+use thiserror::Error;
+
+use crate::directory_sync::{DirectoryGroup, DirectoryGroupId, DirectorySync};
+use crate::{WorkOsError, WorkOsResult};
+
+#[derive(Debug, Error)]
+pub enum GetDirectoryError {}
```
Comment (maxdeviant) [inline src/directory_sync/operations/get_directory_group.rs]:
```suggestion
pub enum GetDirectoryGroupError {}
```

Comment (sheldonvaughn) [reply src/directory_sync/operations/get_directory_group.rs]: Updated all of these, thanks for catching!

**Review by maxdeviant — APPROVED** [review]

File: src/directory_sync/operations/get_directory_group.rs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        Self::Operation(err)
+    }
+}
+
+#[async_trait]
+pub trait GetDirectoryGroup {
+    /// Retrieves a [`DirectoryGroup`] by its ID.
+    ///
+    /// [WorkOS Docs: Get a Directory Group](https://workos.com/docs/reference/directory-sync/group/get)
+    async fn get_directory(
```
Comment (maxdeviant) [inline src/directory_sync/operations/get_directory_group.rs]:
```suggestion
    async fn get_directory_group(
```

**Review by maxdeviant — APPROVED** [review]

**Comment by sheldonvaughn** [comment]
bors merge

**Pull request closed without merging**